### ISSUE TITLE
Couldn’t replicate Mass-DM-Go `Server Joiner` in Python. Need a lot of help!

### ISSUE THREAD

**Issue by muhitrhn** [V4NSH4J/discord-mass-DM-GO#408]
Okay, so as far I think, I have completely replicated the Server Joiner system of DMDGO. The problem is, `I get Captcha when joining with my tool and the tokens get banned after 2 joins` while with the same list of tokens, `MDMGO directly joins the server without any captcha (I’m using same proxy and captcha solver in both)`. I think I missed to replicate something which is important for Server Joiner but couldn’t find anything.

The current python script does -
1. Get Cookies from discord.com (Proxy, same header’s as MDMGO)
2. Get fingerprints (Same proxy, same header’s as MDMGO)
3. Get X-Context (Same proxy, same header’s as MDMGO)
4. Post request to the server join api with invite, token and all other info as MDMGO (Same proxy, same header’s as MDMGO)
5. If got captcha, solve it and resend the solved key with proper headers used in MDMGO (Same proxy)

Did I miss anything which is needed to complete the script? I tried everything I know but still tokens get’s banned after 2 joins. Could you help me?
TIA!

**Comment by foxzsz** [comment]
Have you also tried replicating the TlsClientConfig in DMDGO? Discord happens to check the Tls configuration frequently to filter out bots

Code snippet:
```{
			TLSClientConfig: &tls.Config{
				MinVersion:         tls.VersionTLS12,
				CipherSuites:       []uint16{0x1301, 0x1303, 0x1302, 0xc02b, 0xc02f, 0xcca9, 0xcca8, 0xc02c, 0xc030, 0xc00a, 0xc009, 0xc013, 0xc014, 0x009c, 0x009d, 0x002f, 0x0035},
				InsecureSkipVerify: true,
				CurvePreferences:   []tls.CurveID{tls.CurveID(0x001d), tls.CurveID(0x0017), tls.CurveID(0x0018), tls.CurveID(0x0019), tls.CurveID(0x0100), tls.CurveID(0x0101)},
			```

**Comment by muhitrhn** [comment]
> Have you also tried replicating the TlsClientConfig in DMDGO? Discord happens to check the Tls configuration frequently to filter out bots
> 
> Code snippet:
> 
> ```
> 			TLSClientConfig: &tls.Config{
> 				MinVersion:         tls.VersionTLS12,
> 				CipherSuites:       []uint16{0x1301, 0x1303, 0x1302, 0xc02b, 0xc02f, 0xcca9, 0xcca8, 0xc02c, 0xc030, 0xc00a, 0xc009, 0xc013, 0xc014, 0x009c, 0x009d, 0x002f, 0x0035},
> 				InsecureSkipVerify: true,
> 				CurvePreferences:   []tls.CurveID{tls.CurveID(0x001d), tls.CurveID(0x0017), tls.CurveID(0x0018), tls.CurveID(0x0019), tls.CurveID(0x0100), tls.CurveID(0x0101)},
> 			```
> ```

Ow, that’s something I didn’t add. Have no idea how to do that in python.

**Comment by V4NSH4J** [comment]
Yeah, it is probably your request libraries default TLS config which is causing you to be flagged. You should check where you can set your TLS version and ciphersuits. I'm using values from my browser's fingerprint. Discord is checking your TLS version (should be preferably 1.2 or 1.3) and your ciphersuits (mentioning only the one highest on both client and server list won't work as they're checking the other ones too probably to see if it's a legitimate client). 
Fun thing is, in GO if you used the default client, Discord wouldn't flag you, but if you used proxies in transport, Discord would start to flag you. I'm not exactly sure why that happens but you could try without a proxy to check if it's the same case for you (as a quick means to test if the TLS config is causing problems) Also check your libraries' default setting, if it isn't ideal like I described before, you would have to find a way to set it or use a different library.

**Comment by muhitrhn** [comment]
Okay. I'm on it.

**Comment by muhitrhn** [comment]
@V4NSH4J Btw. How did you get these data?
```
TLSClientConfig: &tls.Config{
		MinVersion:         tls.VersionTLS12,
		CipherSuites:       []uint16{0x1301, 0x1303, 0x1302, 0xc02b, 0xc02f, 0xcca9, 0xcca8, 0xc02c, 0xc030, 0xc00a, 0xc009, 0xc013, 0xc014, 0x009c, 0x009d, 0x002f, 0x0035},
		InsecureSkipVerify: true,
		CurvePreferences:   []tls.CurveID{tls.CurveID(0x001d), tls.CurveID(0x0017), tls.CurveID(0x0018), tls.CurveID(0x0019), tls.CurveID(0x0100), tls.CurveID(0x0101)},

**Comment by V4NSH4J** [comment]
> @V4NSH4J Btw. How did you get these data?
> 
> ```
> TLSClientConfig: &tls.Config{
>         MinVersion:         tls.VersionTLS12,
>         CipherSuites:       []uint16{0x1301, 0x1303, 0x1302, 0xc02b, 0xc02f, 0xcca9, 0xcca8, 0xc02c, 0xc030, 0xc00a, 0xc009, 0xc013, 0xc014, 0x009c, 0x009d, 0x002f, 0x0035},
>         InsecureSkipVerify: true,
>         CurvePreferences:   []tls.CurveID{tls.CurveID(0x001d), tls.CurveID(0x0017), tls.CurveID(0x0018), tls.CurveID(0x0019), tls.CurveID(0x0100), tls.CurveID(0x0101)},
> }
> ```

Checked my browser's TLS Config from a random website like [this one](https://browserleaks.com/ssl)

**Comment by muhitrhn** [comment]
> > @V4NSH4J Btw. How did you get these data?
> > ```
> > TLSClientConfig: &tls.Config{
> >         MinVersion:         tls.VersionTLS12,
> >         CipherSuites:       []uint16{0x1301, 0x1303, 0x1302, 0xc02b, 0xc02f, 0xcca9, 0xcca8, 0xc02c, 0xc030, 0xc00a, 0xc009, 0xc013, 0xc014, 0x009c, 0x009d, 0x002f, 0x0035},
> >         InsecureSkipVerify: true,
> >         CurvePreferences:   []tls.CurveID{tls.CurveID(0x001d), tls.CurveID(0x0017), tls.CurveID(0x0018), tls.CurveID(0x0019), tls.CurveID(0x0100), tls.CurveID(0x0101)},
> > }
> > ```
> 
> Checked my browser's TLS Config from a random website like [this one](https://browserleaks.com/ssl)
> 
> The insecure skip verify makes no difference, it's technically only for http websites and discord forces https

And the `CurvePreferences`?

**Comment by V4NSH4J** [comment]
> > > @V4NSH4J Btw. How did you get these data?
> > > ```
> > > TLSClientConfig: &tls.Config{
> > >         MinVersion:         tls.VersionTLS12,
> > >         CipherSuites:       []uint16{0x1301, 0x1303, 0x1302, 0xc02b, 0xc02f, 0xcca9, 0xcca8, 0xc02c, 0xc030, 0xc00a, 0xc009, 0xc013, 0xc014, 0x009c, 0x009d, 0x002f, 0x0035},
> > >         InsecureSkipVerify: true,
> > >         CurvePreferences:   []tls.CurveID{tls.CurveID(0x001d), tls.CurveID(0x0017), tls.CurveID(0x0018), tls.CurveID(0x0019), tls.CurveID(0x0100), tls.CurveID(0x0101)},
> > > }
> > > ```
> > 
> > 
> > Checked my browser's TLS Config from a random website like [this one](https://browserleaks.com/ssl)
> > The insecure skip verify makes no difference, it's technically only for http websites and discord forces https
> 
> And the `CurvePreferences`?

Same website, check the support groups.

**Comment by muhitrhn** [comment]
Okay. Let me try to find a way to achieve that in python.

**Comment by foxzsz** [comment]
Have a look at this repository https://github.com/VeNoMouS/cloudscraper/tree/master/cloudscraper if you are still figuring it out.

**Comment by itschasa** [comment]
did anyone find a way to do this in python?

**Comment by V4NSH4J** [comment]
> did anyone find a way to do this in python? wondering if discord still check TLS

Discord does still check TLS.

**Comment by V4NSH4J** [comment]
Hey, please re-open this or open another issue for more questions.
